File: app/internal/publisher/template/java/widgets/ActivationButtonWidget.java
package com.websmith.sdk.widgets;

import com.websmith.sdk.ApiClient;
import com.websmith.sdk.CacheManager;
import com.websmith.sdk.HardwareDetector;
import com.google.gson.JsonObject;
import com.google.gson.JsonParser;
import java.io.File;
import java.io.FileReader;
import java.util.Scanner;

public class ActivationButtonWidget {
    public static void render() {
        System.out.println("\n--- Activation Widget ---\n");
        JsonObject config = loadConfig();
        if (config == null || config.entrySet().isEmpty()) {
            System.out.println("Error: api-config.json not found.");
            return;
        }
        CacheManager cache = new CacheManager(config);
        HardwareDetector hardware = new HardwareDetector();
        String hardwareId;
        try {
            hardwareId = hardware.getFingerprint();
            System.out.println("Hardware ID: " + hardwareId + "\n");
        } catch (Exception e) {
            System.out.println("Hardware detection failed: " + e.getMessage());
            return;
        }
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter License Key: ");
        String licenseKey = scanner.nextLine().trim();
        if (licenseKey.isEmpty()) {
            System.out.println("License key is required.");
            return;
        }
        try {
            ApiClient client = new ApiClient(config, hardware, cache);
            System.out.println("Activating license...");
            JsonObject result = client.activateLicense(licenseKey, hardwareId);
            if ((result.has("success") && result.get("success").getAsBoolean())
                || (result.has("data") && result.getAsJsonObject("data").has("success")
                    && result.getAsJsonObject("data").get("success").getAsBoolean())) {
                System.out.println("License activated successfully!");
                cache.invalidateLicenseStatus();
            } else {
                String err = result.has("message") ? result.get("message").getAsString()
                    : result.has("error") ? result.get("error").getAsString()
                    : "Activation failed";
                System.out.println("Failed: " + err);
            }
        } catch (Exception e) {
            System.out.println("Error: " + e.getMessage());
        }
    }

    private static JsonObject loadConfig() {
        String[] paths = {"config/api-config.json", "config/api-config.json"};
        for (String path : paths) {
            try {
                File f = new File(path);
                if (f.exists()) {
                    try (FileReader reader = new FileReader(f)) {
                        return JsonParser.parseReader(reader).getAsJsonObject();
                    }
                }
            } catch (Exception ignored) {}
        }
        return null;
    }
}

File: app/internal/publisher/template/dotnet/widgets/ActivationButtonWidget.cs
using System;
using System.IO;
using System.Text.Json;
using System.Threading.Tasks;

namespace WebsmithSDK.Widgets;

public static class ActivationButtonWidget
{
    public static void Render()
    {
        Console.WriteLine("\n--- Activation Widget ---\n");
        var config = LoadConfig();
        if (config == null)
        {
            Console.WriteLine("Error: api-config.json not found.");
            return;
        }
        var cache = new CacheManager(config);
        var hardware = new HardwareDetector();
        string hardwareId;
        try
        {
            hardwareId = hardware.GetFingerprint();
            Console.WriteLine($"Hardware ID: {hardwareId}\n");
        }
        catch (Exception e)
        {
            Console.WriteLine($"Hardware detection failed: {e.Message}");
            return;
        }
        Console.Write("Enter License Key: ");
        var licenseKey = Console.ReadLine()?.Trim();
        if (string.IsNullOrEmpty(licenseKey))
        {
            Console.WriteLine("License key is required.");
            return;
        }
        try
        {
            var client = new ApiClient(config, hardware, cache);
            Console.WriteLine("Activating license...");
            var result = client.ActivateLicense(licenseKey, hardwareId).Result;
            if ((result.RootElement.TryGetProperty("success", out var s) && s.GetBoolean())
                || (result.RootElement.TryGetProperty("data", out var d)
                    && d.TryGetProperty("success", out var s2) && s2.GetBoolean()))
            {
                Console.WriteLine("License activated successfully!");
                cache.InvalidateLicenseStatus();
            }
            else
            {
                var err = result.RootElement.TryGetProperty("message", out var m)
                    ? m.GetString()
                    : result.RootElement.TryGetProperty("error", out var e)
                        ? e.GetString() : "Activation failed";
                Console.WriteLine($"Failed: {err}");
            }
        }
        catch (Exception e)
        {
            Console.WriteLine($"Error: {e.Message}");
        }
    }

    private static JsonDocument? LoadConfig()
    {
        var paths = new[] { "config/api-config.json", "config/api-config.json" };
        foreach (var path in paths)
        {
            if (File.Exists(path))
            {
                try
                {
                    var text = File.ReadAllText(path);
                    return JsonDocument.Parse(text);
                }
                catch { }
            }
        }
        return null;
    }
}
